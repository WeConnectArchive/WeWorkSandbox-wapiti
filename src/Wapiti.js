import React, {Component} from 'react';
import cx from 'classnames';
import DangerousHTML from 'react-dangerous-html';
import marked from 'marked';
import {ProcessSpec,processGrpcSpec} from './utils/parse-utils.js';
import {EndPoints} from './EndPoints';

import stylef from './FontStyles.scss';
import style from './Wapiti.scss';

export class Wapiti extends Component {

  constructor(props) {
    super(props);
    this.state = {};
  }

  loadSpec(specUrl, specType) {
    let me = this;
    if (!specUrl){
      return;
    }

    this.setState({
      isLoading: true,
      selectedServer: ""
    });

    if (specType === 'rest') {
       ProcessSpec(specUrl).then(function(spec){
        me.setState({
          isLoading: false,
          resolvedSpec: spec
        });
        if (spec===undefined || spec === null){
          console.error('Unable to resolve the API spec. ');
          me.setState({
            isLoading: false,
            errorSpec: "Unable to resolve the REST API spec"
          });          
        }
      })
      .catch(function(err) {
        me.setState({
          isLoading: false,
          errorSpec: "Could not load the REST API spec"
        });
        console.error('Unable to resolve the API spec.. ' + err.message);
      });
    }
    else if (specType === 'grpc') {
      processGrpcSpec(specUrl).then(function(spec) {
        me.setState({
          isLoading: false,
          resolvedSpec: spec
        });
        if (spec===undefined || spec === null){
          console.error('Unable to resolve the GRPC API spec. ');
          me.setState({
            isLoading: false,
            errorSpec: "Unable to resolve the GRPC API spec"
          });
        }
      })
      .catch(function(err) {
        me.setState({
          isLoading: false,
          errorSpec: "Could not load the GRPC API spec"
        });
        console.error('Unable to resolve the GRPC spec : ' + err.message);
      });
    }
  }

  componentDidMount() {
    this.loadSpec(this.props.specUrl, this.props.specType);
  }

  render() {
    return (
      <div className={cx(style["body-container"],stylef["regular-font"])}>
        {this.state.isLoading && (<div style={{textAlign: 'center', margin: '16px'}}>Loading ... </div>)}
        {!this.state.isLoading && this.state.errorSpec && (<div style={{textAlign: 'center', margin: '16px'}}>{this.state.errorSpec} </div>)}
        {this.state.resolvedSpec && this.state.resolvedSpec.info && (
        <div className={cx(style["section-gap"])}>
          <div className={cx(stylef["title"])}>
            {this.state.resolvedSpec.info.title}
            {this.state.resolvedSpec.info.version && (
              <span style={{ fontSize: 'var(--small-font-size)', fontWeight: 'bold'  }}>
                {this.state.resolvedSpec.info.version}
              </span>
            )}
          </div>
          {this.state.resolvedSpec.info.description && (
              <div className={cx(stylef['m-markdown'], stylef["regular-font"])}><DangerousHTML html={marked(this.state.resolvedSpec.info.description)} /></div>
          )}
          </div>
        )}
        {/*TODO SCheme */}
        {this.state.resolvedSpec && this.state.resolvedSpec.tags && (
            <React.Fragment>
            {this.state.resolvedSpec.tags.map( tag => (
            <React.Fragment>
            <div className={cx(stylef["sub-title"],style["tag"],style["section-gap"],stylef["regular-font"])}>{tag.name}</div>
            <div style={{ margin:'4px 20px'}}>
              <div className={cx(stylef["m-markdown"],stylef["regular-font"])}>{tag.description?<DangerousHTML html={marked(tag.description)}/>:''}</div>
            </div>
            <EndPoints
              selectedServer  = "{this.state.selectedServer?this.state.selectedServer:''}"
              paths           = {tag.paths}
              specType        = {this.props.specType}
            ></EndPoints>
            </React.Fragment>
           ))}
           </React.Fragment>
        )}        
      </div>
    );
  }
}
