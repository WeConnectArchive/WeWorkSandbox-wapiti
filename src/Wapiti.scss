.body-container{
  margin:0;

  --bg:#fff;
  --bg2:#fafafa;
  --fg:#333;
  --fg2:#565656;
  --light-fg:#999;
  --very-light-fg:#bbb;
  --pre-border-color:#000;
  --pre-fg:#ccc;
  --pre-bg:#263238;
  --code-fg:#ccc;
  --code-bg:transparent;
  --border-color:#ccc;
  --input-bg:#fff;
  --input-border-color:#C5D9E8;
  --placeholder-color:#dedede;
  --light-border-color:#eee;
  --light-get-color:#eff8fd;
  --light-put-color:#fff5e6;
  --light-post-color:#fbfff0;
  --light-delete-color:#fff0f0;
  --light-patch-color:#fff5cc;
  --hover-color:#f7f7f7;
  --error-color:#ff3333;
  --success-color:#47AFE8;
  --hover-bg:#f7f7f7;
  --get-color:#47AFE8;
  --put-color:#FF9900;
  --post-color:#99CC00;
  --delete-color:#F06560;
  --patch-color:#fc0;
  --link-color:#47AFE8;
  --primary-color:#FF791A;
  --dark-primary-color:darken(#FF791A, 30%);
  --primary-text:#FFFFFF;
  --header-bg:#444;
  --header-fg:#CCC;
  --layout:column;
  --font-mono:Monaco, 'Andale Mono', 'Roboto Mono', Consolas;
  --font-mono-size:14px;
  --font-regular: Helvetica, Arial;
  --title-font-size:16px;
  --regular-font-size:14px;
  --small-font-size:12px;
  --border-radius:2px;

  display:block;
  min-width:375px;
  width:100%;
  height:100%;
  margin:0;
  padding:0;
  overflow: auto;
  letter-spacing:normal;
  color:var(--fg);
  background-color:var(--bg);
  font-family:var(--font-regular);
}

        .section-gap {
          padding: 24px 8px 8px 8px;
        }

        .logo {
          height:36px;
          width:36px;
          margin-left:5px;
        }
        .only-large-screen-flex,
        .only-large-screen{
          display:none;
        }
        .header-title{
          font-size:calc(var(--title-font-size) + 8px); padding:0 8px;
        }
        .tag{
          text-transform: uppercase;
        }

        @media only screen and (min-width: 768px){
          .only-large-screen{
            display:block;
          }
          .only-large-screen-flex{
            display:flex;
          }
          .body-container{
            margin:0 16px;
          }
          .section-gap {
            padding: 24px 24px 8px 24px;
          }
        }
