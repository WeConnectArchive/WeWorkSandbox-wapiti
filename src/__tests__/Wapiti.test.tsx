import * as React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Wapiti } from '../Wapiti';

const theSpec = "https://raw.githubusercontent.com/OAI/OpenAPI-Specification/master/examples/v2.0/yaml/petstore.yaml"
const theType = "rest"

describe('Wapiti', () => {
  it('renders with rest', () => {
    let wrapper = shallow(<Wapiti
          specUrl={theSpec}
          specType={theType}
        />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
