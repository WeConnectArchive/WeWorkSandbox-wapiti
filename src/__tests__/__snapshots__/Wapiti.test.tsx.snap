// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wapiti renders with rest 1`] = `
<div
  className="body-container regular-font"
>
  <div
    style={
      Object {
        "margin": "16px",
        "textAlign": "center",
      }
    }
  >
    Loading ... 
  </div>
</div>
`;
